apicast/src/provider.lua

```diff
@@ -378,20 +381,30 @@ function _M.post_action()
 
   _M.call(service_id) -- initialize resolver and get backend upstream peers
 
-  local method, path, headers = ngx.req.get_method(), ngx.var.request_uri, ngx.req.get_headers()
+  local request_log = {}
+
+  if request_logs then
+    local method, path, headers = ngx.req.get_method(), ngx.var.request_uri, ngx.req.get_headers()
```

Comment:
Consider if extracting two methods to calculate `req` and `res` would make the code easier to understand.